src/osip-simulation-view/src/main/java/edu/kit/pse/osip/simulation/view/main/SimulationMainWindow.java

```diff
@@ -45,32 +213,76 @@ public final void showOverflow() {
+    public final void setControlButtonHandler(AbstractMenuControlButton controlButtonHandler) {
+        menu.setControlButtonHandler(controlButtonHandler);
     }
+
     /**
      * Sets the handler for pressing the settings entry in the menu
      * @param settingsButtonHandler The handler to be called when the settings button is pressed
      */
     public final void setSettingsButtonHandler(AbstractMenuSettingsButton settingsButtonHandler) {
-        throw new RuntimeException("Not implemented!");
+        menu.setSettingsButtonHandler(settingsButtonHandler);
     }
+
     /**
      * Sets the handler for pressing the about entry in the menu
      * @param aboutButtonHandler The handler to be called when the about button is pressed
      */
-    public final void setAboutButtonHandler(MenuAboutButtonHandler aboutButtonHandler) {
-        throw new RuntimeException("Not implemented!");
+    public final void setAboutButtonHandler(AbstractMenuAboutButton aboutButtonHandler) {
+        menu.setAboutButtonHandler(aboutButtonHandler);
     }
+
     /**
      * Sets the handler for pressing the help entry in the menu
      * @param helpButtonHandler The handler to be called when the help button is pressed
      */
-    public final void setHelpButtonHandler(MenuHelpButtonHandler helpButtonHandler) {
-        throw new RuntimeException("Not implemented!");
+    public final void setHelpButtonHandler(AbstractMenuHelpButton helpButtonHandler) {
+        menu.setHelpButtonHandler(helpButtonHandler);
+    }
+
+    /**
+     * This method creates two ChangeListeners that keep track of the window width and height
+     * and, if it changes, change the canvas size accordingly
+     */
+    private void setResizeListeners(Stage primaryStage) {
+
+        final ChangeListener<Number> listener = new ChangeListener<Number>() {
+            final Timer timer = new Timer();
+            TimerTask task = null;
+            final long delayTime = 50;
+
+            @Override
+            public void changed(ObservableValue<? extends Number> observable, Number oldValue, final Number newValue) {
+                if (task != null) { // there was already a task scheduled from the previous operation
+                    task.cancel(); // cancel it, we have a new size to consider
+                }
+
+                task = new TimerTask() {
+                    @Override
+                    public void run() {
+                        synchronized (canvas) {
+
+                            double newWidth = primaryStage.getWidth();
+                            double newHeight = primaryStage.getHeight() - 50;
+
+                            canvas.setWidth(newWidth);
+                            canvas.setHeight(newHeight);
+                            canvas.getGraphicsContext2D().clearRect(0, 0, newWidth, newHeight);
+                        }
+                    }
+                };
+
+                timer.schedule(task, delayTime);
```

Comment:
Es wird delayTime (50) ms gewartet, bis der resize stattfindet. Sonst spielt es komplett verrueckt, wenn es immer sofort updaten will.
Kommt in der Zeitspanne eine weitere Aenderung wird wieder neu mit warten angefangen.